cplusplus/VImage.cpp

```diff
@@ -767,8 +767,9 @@ std::vector<VImage>
 VImage::bandsplit( VOption *options ) const
 {
 	std::vector<VImage> b;
+	b.reserve(bands());
 
-	for( int i = 0; i < bands(); i++ )
+	for( int i = 0; i < int(b.capacity()); i++ )
```

Comment:
As @lovell says, this should probably stay as `bands()` since vectors are allowed to over-allocate.